dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -58,6 +63,17 @@ def refute_estimate(self):
                 else:
                     new_outcome = self._data[self._outcome_name].sample(frac=1,
                                                                 random_state=self._random_state).values
+            elif self._outcome_function is not None:
+                new_outcome = self._outcome_function(self._data)
+                if len(new_outcome.shape) == 2 and \

```

Comment:
Trying to understand the if-conditions. When is the first condition not sufficient?

Reply:
It accounts for functions that pass row or column vectors. Rather than throwing an error on a shape mismatch it convert it to a 1 d vector
(n,1) or (1, n) to (n,) with the help of the ravel function
The second condition is for 1d arrays, which is ideally what we want.

Reply:
I see, so this condition " ( new_outcome.shape[0] ==1 or new_outcome.shape[1] ):" is to check if it is 1D array data?

Reply:
That is to make sure either the number of rows or columns is 1, so that we have a row or column  vector.